stack.go

```diff
@@ -6,47 +6,80 @@ import (
 	"strings"
 )
 
+// Stack is an array of stack frames stored in a human readable format.
+type Stack []StackFrame
+
+// contains checks if a stack trace contains a specific frame.
+func (s *Stack) contains(frame StackFrame) bool {
+	for _, f := range *s {
+		if f == frame {
+			return true
+		}
+	}
+	return false
+}
+
+// insertFrames inserts a wrap error frame into the correct place of the root error stack trace.
+func (s *Stack) insertFrame(wFrames []StackFrame) {
+	if len(wFrames) == 0 || s.contains(wFrames[0]) {
+		return
+	} else if len(wFrames) == 1 {
+		// append the frame to the end if there's only one
+		*s = append(*s, wFrames[0])
+	}
+
+	rStack := *s
+	for i, f := range rStack {
+		if f == wFrames[1] {
+			// insert the first frame into the stack if the second frame is found
+			rStack = append(rStack[:i], append([]StackFrame{wFrames[0]}, rStack[i:]...)...)
```

Comment:
this is not readable, suggest we break it down for better readability.

Reply:
also providing an example will help readers tremendously on how this is being done.

Reply:
unfortunately, this is pretty standard syntax for inserting into a slice 😂but i'll see if i can make it more clear. it would be nice if Go had an insert method (similar to append).